src/Middleware/CORS/src/Infrastructure/CorsPolicy.cs

```diff
@@ -71,10 +72,26 @@ public bool AllowAnyOrigin
             }
         }
 
+        /// <summary>
+        /// Gets a value indicating if <see cref="IsOriginAllowed"/> is the default function that is set in the CorsPolicy constructor.
+        /// </summary>
+        public bool HasDefaultIsOriginAllowed { get; private set; } = true;
```

Comment:
> Could you make this `internal`?

Since this property affects behaviour and there is no other way to compute the value without access to the private `DefaultIsOriginAllowed`, I thought making this property `public` would be appropriate. But I can make it `internal` of course.

> _isOriginAllowed == DefaultIsOriginAllowed
> Does this work?

I introduced the boolean in lieu of comparing the delegate because I was concerned about performance. However, I didn't actually check the generated assembly to substantiate my concern. I'll do that this weekend.

Reply:
I've checked with the .NET 5.0.100-preview.5.20272.9 JIT on Windows (with disabled tiered compilation) that in the code generated for the delegate equality comparison a delegate is allocated on the heap (with CORINFO_HELP_NEWSFAST). Hence, my implementation using a stored boolean currently seems preferable for performance reasons.

Reply:
@stephan-tolksdorf sorry I missed this comment. Can we make this a internal bool? This does not require a public API change.